examples/C++/HelloWorldExampleSharedMem/HelloWorld.cxx

```diff
@@ -0,0 +1,260 @@
+
+
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+
+
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4) + data.message().size() + 1;
+
+    current_alignment += ((DATA_SIZE) * 1) + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+
+
+    return current_alignment - initial_alignment;
+}
+
+void HelloWorld::serialize(eprosima::fastcdr::Cdr &scdr) const
+{
+
+    scdr << m_index;
+    scdr << m_message;
+    scdr << m_data;
+
+}
+
+void HelloWorld::deserialize(eprosima::fastcdr::Cdr &dcdr)
+{
+
+    dcdr >> m_index;
+    dcdr >> m_message;
+    dcdr >> m_data;
+
+}
+
+/*!
+ * @brief This function sets a value in member index
+ * @param _index New value for member index
+ */
+void HelloWorld::index(uint32_t _index)
+{
+m_index = _index;
+}
+
+/*!
+ * @brief This function returns the value of member index
+ * @return Value of member index
+ */
+uint32_t HelloWorld::index() const
+{
+    return m_index;
+}
+
+/*!
+ * @brief This function returns a reference to member index
+ * @return Reference to member index
+ */
+uint32_t& HelloWorld::index()
+{
+    return m_index;
+}
+
+/*!
+ * @brief This function copies the value in member message
+ * @param _message New value to be copied in member message
+ */
+void HelloWorld::message(const std::string &_message)
+{
+m_message = _message;
+}
+
+/*!
+ * @brief This function moves the value in member message
+ * @param _message New value to be moved in member message
+ */
+void HelloWorld::message(std::string &&_message)
+{
+m_message = std::move(_message);
+}
+
+/*!
+ * @brief This function returns a constant reference to member message
+ * @return Constant reference to member message
+ */
+const std::string& HelloWorld::message() const
+{
+    return m_message;
+}
+
+/*!
+ * @brief This function returns a reference to member message
+ * @return Reference to member message
+ */
+std::string& HelloWorld::message()
+{
+    return m_message;
+}
+/*!
+ * @brief This function copies the value in member data
+ * @param _data New value to be copied in member data
+ */
+void HelloWorld::data(const std::array<char, DATA_SIZE> &_data)
+{
+m_data = _data;
+}
+
+/*!
+ * @brief This function moves the value in member data
+ * @param _data New value to be moved in member data
+ */
+void HelloWorld::data(std::array<char, DATA_SIZE> &&_data)
+{
+m_data = std::move(_data);
+}
+
+/*!
+ * @brief This function returns a constant reference to member data
+ * @return Constant reference to member data
+ */
+const std::array<char, DATA_SIZE>& HelloWorld::data() const
+{
+    return m_data;
+}
+
+/*!
+ * @brief This function returns a reference to member data
+ * @return Reference to member data
+ */
+std::array<char, DATA_SIZE>& HelloWorld::data()
+{
+    return m_data;
+}
+
+size_t HelloWorld::getKeyMaxCdrSerializedSize(size_t current_alignment)
+{
+	size_t current_align = current_alignment;
+
+
+
+
+
+
+    return current_align;
+}
```

Comment:
Implementation missing? If it waiting to be developed in the future, add a TODO to keep track.